caas/kubernetes/provider/builtin.go

```diff
@@ -69,24 +72,18 @@ func getLocalMicroK8sConfig(cmdRunner CommandRunner) ([]byte, error) {
 	if err != nil {
 		return []byte{}, errors.NotFoundf("microk8s")
 	}
-	execParams := exec.RunParams{
-		Commands: "microk8s config",
+	notSupportErr := errors.NewNotSupported(nil, fmt.Sprintf("juju %q can only work with strict confined microk8s", version.Current))
```

Comment:
SNAP_DATA is a juju snap env var; we are not dealing with microk8s at this point
We also need to support the original shell out to microk8s.config so that juju can be run be developers outside the snap. So if SNAP_DATA is set, we need to do things using the content interface, if not, we can still use the original way

Reply:
As discussed, keep consistency, we read the kubeconfif file from `/var/snap/microk8s/current/credentials/client.config` directly instead of using `microk8s.config` for the local dev environment.